src/core/matrix.c

```diff
@@ -125,19 +168,88 @@ igraph_error_t igraph_matrix_complex_realimag(const igraph_matrix_complex_t *v,
     return IGRAPH_SUCCESS;
 }
 
+/**
+ * \ingroup matrix
+ * \function igraph_matrix_complex_create
+ * \brief Creates a complex matrix from a real and imaginary part.
+ *
+ * \param v Pointer to an uninitialized complex matrix.
+ * \param real Pointer to the real part of the complex matrix.
+ * \param imag Pointer to the imaginary part of the complex matrix.
+ * \return Error code.
+ *
+ * Time complexity: O(n),
+ * n is the
+ * number of elements in the matrix.
+ */
+
 igraph_error_t igraph_matrix_complex_create(igraph_matrix_complex_t *v,
                                  const igraph_matrix_t *real,
                                  const igraph_matrix_t *imag) {
-    IGRAPH_CHECK(igraph_vector_complex_create(&v->data, &real->data,
-                 &imag->data));
+    igraph_integer_t nrowr = igraph_matrix_nrow(real);
+    igraph_integer_t ncolr = igraph_matrix_ncol(real);
+    igraph_integer_t nrowi = igraph_matrix_nrow(imag);
+    igraph_integer_t ncoli = igraph_matrix_ncol(imag);
+
+    if (nrowr != nrowi) {
+        IGRAPH_ERRORF("Number of rows in real matrix (%" IGRAPH_PRId
+                ") not equal to number of rows in imaginary matrix (%"
+                IGRAPH_PRId ").", IGRAPH_EINVAL, nrowr, nrowi);
+    }
+    if (ncolr != ncoli) {
+        IGRAPH_ERRORF("Number of columns in real matrix (%" IGRAPH_PRId
+                ") not equal to number of columns in imaginary matrix (%"
+                IGRAPH_PRId ").", IGRAPH_EINVAL, ncolr, ncoli);
+    }
+
+    igraph_matrix_complex_init(v, nrowr, ncolr);
+
+    for (igraph_integer_t i = 0; i < nrowr * ncolr; i++) {
+        VECTOR(v->data)[i] = igraph_complex(VECTOR(real->data)[i], VECTOR(imag->data)[i]);
+    }
+
     return IGRAPH_SUCCESS;
 }
 
+/**
+ * \ingroup matrix
+ * \function igraph_matrix_complex_create_polar
+ * \brief Creates a complex matrix from a magnitude and an angle.
+ *
+ * \param v Pointer to an uninitialized complex matrix.
+ * \param r Pointer to the magnitude of the complex matrix.
+ * \param theta Pointer to the angle of the complex matrix.
+ * \return Error code.
+ *
+ * Time complexity: O(n),
+ * n is the
+ * number of elements in the matrix.
+ */
+
 igraph_error_t igraph_matrix_complex_create_polar(igraph_matrix_complex_t *v,
                                        const igraph_matrix_t *r,
                                        const igraph_matrix_t *theta) {
-    IGRAPH_CHECK(igraph_vector_complex_create_polar(&v->data, &r->data,
-                 &theta->data));
+    igraph_integer_t nrowr = igraph_matrix_nrow(r);
+    igraph_integer_t ncolr = igraph_matrix_ncol(r);
+    igraph_integer_t nrowi = igraph_matrix_nrow(theta);
+    igraph_integer_t ncoli = igraph_matrix_ncol(theta);
```

Comment:
Nitpick: Can we rename these from `nrowi` to `nrowt` or similar? `r` and `i` look like real / imag.